docs/core/upgrade.md

```diff
@@ -0,0 +1,37 @@
+<!--
+order: 15
+-->
+
+# Upgrades
+
+Upgrade your app modules smoothly with custom in-place migration logic.
+
+The Cosmos SDK currently has two ways to perform upgrades. The first way is by exporting the entire application state to a JSON file, making changes, and then starting a new binary with the changed JSON file as the genesis file. The second way is by performing upgrades in place, significantly decreasing the time needed to perform upgrades for chains with a larger state. The following will guide you on how to setup your application to take advantage of the second method described above.
+
+# Enabling Upgrades
+
+To enable your application to conform to the upgrade module's specifications, a few changes need to be made to your application.
+
+# Genesis State
+
+Each app module's consensus version must be saved to state on the application's genesis. This can be done by adding the following line to the `InitChainer` method in `app.go`
+
+```diff
+func (app *MyApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
+  ...
++ app.UpgradeKeeper.SetModuleVersionMap(ctx, app.mm.GetVersionMap())
+  ...
+}
+```
+
```

Comment:
I think it would be useful to a paragraph here about what VersionMap and ConsensusVersion are